src/plugins/es_ui_shared/static/forms/hook_form_lib/types.ts

```diff
@@ -211,7 +211,7 @@ export interface FieldHook<T = unknown, I = T> {
 export interface FieldConfig<T = unknown, FormType extends FormData = FormData, I = T> {
   readonly label?: string;
   readonly labelAppend?: string | ReactNode;
-  readonly helpText?: string | ReactNode;
+  readonly helpText?: string | ReactNode | (() => ReactNode);
```

Comment:
Here we allow a lazy `helpText`. It is already used like this in some places, it worked because a function could be assigned to `ReactNode`  due to a mistake in @types/react@17. This is no longer allowed with @types/react@18